src/lib/include/vector_triple.h

```diff
@@ -85,38 +90,41 @@ template <typename VectorT> class VectorTriple {
       : x(a), y(b), z(c) {}
 
   // construct by loading from an array of ScalarT eg float* or double *.
-  inline VectorTriple(const ScalarT *source)
-      : x(_genericload<VectorT>(source)),
-        y(_genericload<VectorT>(&source[ValuesPerPack<VectorT>])),
-        z(_genericload<VectorT>(&source[+2 * ValuesPerPack<VectorT>])) {}
-
-  // construct by loading discontiguously from an array of ScalarT eg float* or
-  // double*. Must pass references as deinterleave must happen on x,y and z simultaneously
-  inline VectorTriple(ScalarT *source, const ScalarT *end,
-                      const std::size_t *idxs) {
-                        _genericidxload<VectorT>(source, end, idxs, this->x, this->y, this->z);
+  inline explicit VectorTriple(const ScalarT *source) {
+    if (ValuesPerPack<VectorT> == 1) {
+      x = genericload<VectorT>(source);
+      y = genericload<VectorT>(source + 1);
+      z = genericload<VectorT>(source + 2);
+    } else {
+      auto t1 = genericload<VectorT>(source);
+      auto t2 = genericload<VectorT>(source + ValuesPerPack<VectorT>);
+      auto t3 = genericload<VectorT>(source + ValuesPerPack<VectorT> * 2);
+      Deinterleave3(t1, t2, t3, x, y, z);
```

Comment:
@hmacdope here the default behaviour of the constructor is changed so that you always `.Deinterleave()` because there's not a use case for not

Reply:
Cool beans, I think i get why now.